randommenugen_withextras.rb

```diff
@@ -0,0 +1,58 @@
+#optionals:
+# Expand your solution to ensure that no descriptive term
+# in a menu item is ever repeated.
+
+#DID THIS USING .pop and shuffle!
+
+# So if the first menu item is hot
+# pan-fried dumplings, the the phrases hot, pan-friend and dumpling cannot
+# individually be used in any other menu items.
+#
+# Expand your solution to allow the user to determine how many
+# items they'd like to see via user input.
+# Note: You will need to
+# ensure that this user-chosen number of items is not larger than
+# the number of items you have in your arrays.
+# Instead of using hardcoded or predefined arrays, make your program
+# accept user input. This user input will be utilized to generate
+# the menu items. You'll need to prompt for and store a varying
+# number of entries for each food-type array.
+menu_builder = [[],[],[]]
+menu_builder[0] = %w{Hot Cold Quick Slow Laser-torched Loud Easy Crackling Fresh Soft}
+menu_builder[1] = %w{scrambled fried baked steamed pan-seared burned boiled pickled soaked smoked}
+menu_builder[2] = []
+
+puts "Welcome to Chez Tanya, I'm about to show you the daily menu!\n We get creative up in here."
+puts "Scared? Wanna create your own menu? (Y or N)\n>"
+choice = gets.chomp.upcase!
```

Comment:
The `!` should probably be taken off, since it returns the original letter the user typed and assigns that to choice before the mutation takes effect.